### DISCUSSION TITLE
Conflict between save and write

### DISCUSSION THREAD

**Discussion by inwardmovement** [prettier/prettier#15778] [Q&A] [answered]
After installing next and prettier:

```
npx create-next-app@latest
npm install --save-dev --save-exact prettier
node --eval "fs.writeFileSync('.prettierrc','{}\n')"
```

When I save `tsconfig.json`, it's formatted from this (extract):

`...
"lib": ["dom", "dom.iterable", "esnext"],
...`

to this:

```
...
"lib": [
  "dom",
  "dom.iterable",
  "esnext"
],
...
```

When I run `npx prettier . --write`, it's formatted back to:

`...
"lib": ["dom", "dom.iterable", "esnext"],
...`

Where does the conflict come from, between the format on save and write commands?
Both should use the same local node module and config, so have the same result.
I use VSCode with the Prettier extension.

**Comment by fisker** [comment, marked as answer]
You should config the `editor.defaultFormatter`  https://stackoverflow.com/questions/63954584/how-to-make-prettier-the-default-formatter-in-vs-code

**Reply by inwardmovement** [reply]
Oops I thought I did, but I had this:

```
"[jsonc]": {
  "editor.defaultFormatter": "vscode.json-language-features"
},
```

Thanks!
